text/compare_similarity.ts

```diff
@@ -13,6 +10,14 @@ export interface CompareSimilarityOptions {
    * @default {false}
    */
   caseSensitive?: boolean;
+  /**
+   * Function used to compare two strings.
+   * Lower number means closer match.
+   * 0 means numbers match.
+   *
+   * @default {getWordDistance}
+   */
+  algorithm?: (str1: string, str2: string) => number;
```

Comment:
`algorithm` sounds a bit too generic name for this option to me. How about `distance`?

Reply:
Distance sounds like a variable rather than a function. How about `distanceCalculator`?

Reply:
I feel it's also common to call such function as 'distance' in CS context. For example 'distance' in levenshtein distance, hamming distance, etc

Reply:
Other options could include:
- `distanceAlgorithm`
- `distanceMetric`
- `distanceFunction`
- `distanceFn`

I'm feeling more `distanceFn`. Let me know if this is fine. I'm also not sure if this would be consistent with other the other `std` libraries.

Reply:
`distanceFn` sounds fine to me